browser/ui/webui/cr_components/customize_color_scheme_mode/brave_customize_color_scheme_mode_handler_unittest.cc

```diff
@@ -0,0 +1,181 @@
+
+#include "brave/browser/ui/webui/cr_components/customize_color_scheme_mode/brave_customize_color_scheme_mode_handler.h"
+
+#include <memory>
+
+#include "base/run_loop.h"
+#include "brave/browser/themes/brave_dark_mode_utils.h"
+#include "chrome/browser/themes/theme_service.h"
+#include "chrome/browser/themes/theme_service_factory.h"
+#include "chrome/test/base/scoped_testing_local_state.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "chrome/test/base/testing_profile.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+class MockClient : public customize_color_scheme_mode::mojom::
+                       CustomizeColorSchemeModeClient {
+ public:
+  MockClient() = default;
+  ~MockClient() override = default;
+
+  mojo::PendingRemote<
+      customize_color_scheme_mode::mojom::CustomizeColorSchemeModeClient>
+  BindAndGetRemote() {
+    CHECK(!receiver_.is_bound());
+    return receiver_.BindNewPipeAndPassRemote();
+  }
+
+  MOCK_METHOD(void,
+              SetColorSchemeMode,
+              (customize_color_scheme_mode::mojom::ColorSchemeMode mode),
+              (override));
+
+ private:
+  mojo::Receiver<
+      customize_color_scheme_mode::mojom::CustomizeColorSchemeModeClient>
+      receiver_{this};
+};
+
+class BraveCustomizeColorSchemeModeHandlerUnitTest : public testing::Test {
+ protected:
+  // testing::Test:
+  void SetUp() override {
+    handler_ = std::make_unique<BraveCustomizeColorSchemeModeHandler>(
+        mock_client_.BindAndGetRemote(),
+        mojo::PendingReceiver<customize_color_scheme_mode::mojom::
+                                  CustomizeColorSchemeModeHandler>(),
+        &testing_profile_);
+    dark_mode::SetUseSystemDarkModeEnabledForTest(true);
+  }
+
+  void TearDown() override {
+    handler_.reset();
+    dark_mode::SetUseSystemDarkModeEnabledForTest(false);
+  }
+
+  sync_preferences::TestingPrefServiceSyncable* GetTestingPrefService() {
+    return testing_profile_.GetTestingPrefService();
+  }
+
+  content::BrowserTaskEnvironment task_environment_;
+  ScopedTestingLocalState local_state_{TestingBrowserProcess::GetGlobal()};
+  TestingProfile testing_profile_;
+  testing::NiceMock<MockClient> mock_client_;
+
+  std::unique_ptr<BraveCustomizeColorSchemeModeHandler> handler_;
+};
+
+TEST_F(BraveCustomizeColorSchemeModeHandlerUnitTest,
+       ClientSetColorSchemeModeShouldBeCalledWhenBraveDarkModeTypeChanges) {
+  auto run_loop = std::make_unique<base::RunLoop>();
+  EXPECT_CALL(mock_client_,
+              SetColorSchemeMode(
+                  customize_color_scheme_mode::mojom::ColorSchemeMode::kDark))
+      .WillOnce([&]() { run_loop->Quit(); });
+  dark_mode::SetBraveDarkModeType(
+      dark_mode::BraveDarkModeType::BRAVE_DARK_MODE_TYPE_DARK);
+  run_loop->Run();
```

Comment:
don't need to call `VerifyAndClearExpectations` here?

Reply:
That's new to me. Thanks. 

But I don't think we should use that in this case

From here, https://google.github.io/googletest/gmock_cheat_sheet.html#verifying-and-resetting-a-mock , the expectation will be verified when the mock object is destroyed. Either, we can call that but we shouldn't set new expectations after the verification.

> Do not set new expectations after verifying and clearing a mock after its use

Reply:
I might get it wrong. It could be setting new expectation after `VerifyAndClear()` is prohibited while `VerifyAndClearExpectation` is allowed.

Reply:
@simonhong I updated tests using TestFuture, but not sure if it looks okay. Could you take another look?